src/hexen/p_setup.c

```diff
@@ -801,11 +801,22 @@ void P_SetupLevel(int episode, int map, int playermask, skill_t skill)
     if (i == W_GetNumForName("COLORMAP"))
     {
         LevelUseFullBright = true;
+#ifdef CRISPY_TRUECOLOR
+        actual_colormap = "COLORMAP";
+#endif
     }
     else
     {                           // Probably fog ... don't use fullbright sprites
         LevelUseFullBright = false;
+#ifdef CRISPY_TRUECOLOR
+        // [cirspy] TODO - do not use hardcoded "FOGMAP" lump name.
+        // Need to get it from MAPINFO's fadetable variable, but it's int type there.
+        actual_colormap = "FOGMAP";
```

Comment:
Here's where real problem with custom colormaps comes in. At the moment, `FOGMAP` lump name is hardcoded to provide some compatibility with foggy maps, but in `MAPINFO` lump such name can be different. The problem is - lump name (aka `info->fadetable`) is handled via `int` type, while we need it as externalized `char*`. Not sure what to do yet.

Another idea: use clean and unmodified `R_InitColormaps` for palleted render and modified for true color, as things starts to changing.

Extra idea: if `FOGMAP` is not replaced, try to generate it dynamically. It is impossible to make a clean replica of original map, but at least it should be possible to get a cleaned for since palleted looks a bit awful.

Reply:
FWIW, there are no Hexen wads in the idgames archive that use a custom FOGMAP. Maybe one exists somewhere on the internet but it's not too likely. (AFAICT no pk3s do either though some probably use custom ZDoom functions for fog.)

Reply:
The best news to hear on a Sunday morning! ❤️ Then I still need to investigate how to generate nice-looking fog map. Not sure if it is possible to achieve vanilla effect, but at least it is worth to try, as foggy maps with FOGMAP colormaps looks a yucky comparing to neat non-foggy true colors.